python_modules/dagster/dagster_tests/core_tests/host_representation_tests/test_external_sensor_data.py

```diff
@@ -0,0 +1,51 @@
+from typing import AbstractSet
+
+from dagster import AssetKey, AssetSelection, Definitions, asset, sensor
+from dagster._core.definitions.asset_graph import AssetGraph
+from dagster._core.host_representation.external_data import external_sensor_data_from_def
+
+
+def make_three_assets():
+    @asset
+    def asset1():
+        ...
+
+    @asset
+    def asset2():
+        ...
+
+    @asset
+    def asset3():
+        ...
+
+    return asset1, asset2, asset3
+
+
+def test_external_sensor_has_asset_selection():
+    @sensor(asset_selection=["asset1", "asset2"])
+    def sensor1():
+        ...
+
+    defs = Definitions(sensors=[sensor1])
+
+    assert (
+        external_sensor_data_from_def(sensor1, defs.get_repository_def()).asset_selection
+        == sensor1.asset_selection
+    )
+
+
+def test_unserializable_asset_selection():
```

Comment:
what about `asset_selection=AssetSelection.keys("foo") & MySpecialAssetSelection()`?

In this case, the top level AssetSelection will be `AndAssetSelection`, which is whitelisted for serdes, but the object as a whole will not be possible to serialize.

I think the easiest way to handle this would probably be just a `to_serializable_form()` type of abstract method on the AssetSelection base class

Reply:
oof great point